### ISSUE TITLE
Migrate visual importer to forms, class-based views

### ISSUE THREAD

**Issue by czlee** [czlee/tabbycat#421]
Things that should be done:

- The importer should use class-based views (as part of our more general migration in #228)
- It should use model forms where possible to delegate data validation to models.
- Where appropriate, it should use the standard data importer, but this might not be sensible in most screens here.

Relates to #394, see also #422.

**Comment by czlee** [comment]
Done institutions and teams, current issues:
1. For some reason, if I try to create more than 16 teams or teams from more than 8 institution, the server reports that it processed the request fine, but my browser reports that the connection was reset. I can't figure out why this could possibly be the case. Haven't tried on Heroku.
2. This class-based system is great for institutions, but it turns out the Django model form abstractions don't help that much for teams, I had to do a bit of overriding because the model validation part is a little convoluted 😞 https://github.com/czlee/tabbycat/blob/f463ddb2/tabbycat/importer/forms.py#L139-L174

**Comment by philipbelesky** [comment]
Hmm that limit is super odd, maybe there is some sort of limit on POST data? Not that that makes sense. Have you tried on a different IP address/port perhaps?

**Comment by czlee** [comment]
Okay, I tried it on a different IP address and it work, then I realized the debug toolbar was off, so I tried it on my original IP (localhost) with debug toolbar disabled and it works like a charm. Mystery solved!

**Comment by czlee** [comment]
Spoke too soon, now it fails intermittently, just much less frequently. Weird.

**Comment by philipbelesky** [comment]
@czlee when you have time (but ideally before Wednesday my time :p) could you let me know what needs to be finished here for this branch to be merged? It would be great to have it in place for Easters as I think they wont be using the csv importer.

**Comment by czlee** [comment]
It's just venue constraints left, in order not to cause a regression. Everything else is in place. If you're okay with it being merged without venue constraints, or just leaving the old venue constraints importer in place, there's no reason it can't be merged now.

I haven't removed the old importer yet, so the old URLs will still be accessible, but I was planning on removing those before merging. Let me know if you'd rather leave them in for a while, maybe we should only remove them just before general release.

**Comment by philipbelesky** [comment]
Ok, I have pushed up the new widget. It was relatively easy to get it to use an input with a data list so that it would filter in an autocomplete fashion. To get it to interact with the content type (ie so it would limit choices to adjudicators) I had to add some jQuery to filter this list based on the selected type (all the corresponding options have a suffix like `(Adjudicator)` that this can match against. It's a little brittle but didn't see anyway to do so otherwise.

Will merge into develop but leave the branch and the issue open for you to close if there is anything that needs more polish.
